drivers/gpu/drm/panel/panel-samsung-s6e3fa0.c

```diff
@@ -0,0 +1,741 @@
+{MCS_GAMMA, 0,   221, 0,   227, 0,   217, 131, 130, 130, 131, 131, 131, 131, 132, 131, 131, 132, 133, 129, 128, 128, 128, 129, 128, 131, 133, 132, 134, 132, 133, 0,   0,   0  },
+{MCS_GAMMA, 0,   225, 0,   230, 0,   221, 131, 130, 130, 130, 130, 130, 131, 132, 131, 128, 129, 130, 129, 128, 128, 128, 129, 128, 131, 133, 132, 134, 132, 133, 0,   0,   0  },
+{MCS_GAMMA, 0,   229, 0,   234, 0,   226, 130, 129, 129, 130, 130, 130, 130, 131, 130, 129, 130, 131, 130, 128, 128, 129, 130, 130, 127, 129, 128, 134, 132, 133, 0,   0,   0  },
+{MCS_GAMMA, 0,   234, 0,   238, 0,   231, 130, 129, 129, 130, 130, 130, 130, 131, 131, 129, 130, 131, 128, 127, 127, 127, 128, 127, 127, 129, 128, 134, 132, 133, 0,   0,   0  },
+{MCS_GAMMA, 0,   238, 0,   241, 0,   236, 129, 129, 129, 130, 130, 130, 130, 130, 130, 129, 129, 130, 129, 128, 128, 128, 129, 129, 132, 131, 131, 127, 127, 127, 0,   0,   0  },
+{MCS_GAMMA, 0,   244, 0,   246, 0,   242, 128, 128, 128, 129, 129, 129, 130, 130, 130, 128, 129, 129, 129, 129, 129, 130, 130, 130, 127, 127, 127, 127, 127, 127, 0,   0,   0  },
+{MCS_GAMMA, 0,   247, 0,   249, 0,   246, 129, 129, 129, 129, 129, 129, 129, 129, 129, 128, 128, 128, 130, 129, 129, 127, 127, 127, 127, 127, 127, 127, 127, 127, 0,   0,   0  },
+{MCS_GAMMA, 0,   252, 0,   252, 0,   251, 128, 128, 128, 129, 129, 129, 129, 129, 129, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 0,   0,   0  },
+{MCS_GAMMA, 1,   0,   1,   0,   1,   0,   128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 0,   0,   0  },
+};
+
+/* sequences to lock/unlock Manufacture Time Programming (MTP) commands */
+static const u8 seq_s6e3fa0_test_key_en[6] = {
+	0xf0, 0x5a, 0x5a,  /* level 1 */
+	0xfc, 0x5a, 0x5a   /* level 2 */
+};
+
+static const u8 seq_s6e3fa0_test_key_dis[6] = {
+	0xf0, 0xa5, 0xa5,  /* level 1 */
+	0xfc, 0xa5, 0xa5   /* level 2 */
+};
+
+struct s6e3fa0_sequences_data {
+	const u8 *test_key_enable_seq;
+	const u8 *test_key_disable_seq;
+
+	const u8 (*aid_seq)[NUM_AID_SEQUENCES][AID_SEQUENCE_LEN];
+	const u8 (*elvss_seq)[NUM_ELVSS_SEQUENCES][ELVSS_SEQUENCE_LEN];
+	const u8 (*gamma_seq)[S6E3FA0_NUM_GAMMA_LEVELS][GAMMA_SEQ_LEN];
+};
+
+static const struct s6e3fa0_sequences_data seqdata_s6e3fa0 = {
+	.test_key_enable_seq = seq_s6e3fa0_test_key_en,
+	.test_key_disable_seq = seq_s6e3fa0_test_key_dis,
+	.aid_seq = &seq_s6e3fa0_aid,
+	.elvss_seq = &seq_s6e3fa0_elvss,
+	.gamma_seq = &seq_s6e3fa0_lux
+};
+
+struct s6e3fa0_ctx {
+	struct drm_panel panel;
+	struct mipi_dsi_device *dsi;
+	struct regulator_bulk_data supplies[2];
+	struct gpio_desc *reset_gpio;
+	const struct s6e3fa0_sequences_data *seq_data;
+};
+
+/* helper to simplify code */
+#define dsi_generic_write_seq(dsi, seq...) do {				\
+		static const u8 d[] = { seq };				\
+		int ret;						\
+		ret = mipi_dsi_generic_write(dsi, d, ARRAY_SIZE(d));	\
+		if (ret < 0)						\
+			return ret;					\
+	} while (0)
```

Comment:
in v6.0 (linux-next as for now) there is a standardized macro for this: [mipi_dsi_dcs_write_seq()](https://elixir.bootlin.com/linux/v6.0-rc7/C/ident/mipi_dsi_dcs_write_seq)

I know this MR is for v5.19, which seems to have no `mipi_dsi_dcs_write_seq()`, but something to keep in mind that you'll have to rewrite this block (and all usages of it) during rewrite for 6.0 release. Or if you decide to send it upstream (where you should always base on linux-next), they'll request to change it anyway